service/internal/integrationtest/s3/presign_test.go

```diff
@@ -0,0 +1,83 @@
+// +build integration
+
+package s3
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/internal/integrationtest"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+func TestInteg_PresignURL_PutObject(t *testing.T) {
```

Comment:
Nit would be good to have a PresignURL_PutObject integ test that creates the presign URL without an explicit body.  This also means that the content-length header would not be signed.

```go
params := &s3.PutObjectInput{
     Bucket: &setupMetadata.Buckets.Source.Name,
     Key: &key,
}
```

Looks like this test is almost all setup for it, just need to add the case.

Reply:
Could also have a presigned URL integration test case where the X-Amz-Content-Sha256 header is also provided manually by the user, which would become a signed header.

Reply:
Adding a check against expected set of signed headers would be helpful as well.